units/github_status_api.go

```diff
@@ -138,211 +93,122 @@ func NewGithubStatusUpdateJobForBadConfig(intentID string) amboy.Job {
+			"job":     githubStatusUpdateJobName,
+			"message": "github status updates are disabled, not updating status",
+		})
+		return errors.New("github status updates are disabled, not updating status")
 	}
 
 	return nil
 }
 
-func (j *githubStatusUpdateJob) fetch(status *githubStatus) (err error) {
+func (j *githubStatusUpdateJob) fetch() (*message.GithubStatus, error) {
 	var patchDoc *patch.Patch
-	patchVersion := j.FetchID
+	var err error
+	status := message.GithubStatus{}
 
 	if j.UpdateType == githubUpdateTypeBadConfig {
 		var intent patch.Intent
 		intent, err = patch.FindIntent(j.FetchID, patch.GithubIntentType)
 		if err != nil {
-			return errors.Wrap(err, "can't fetch patch intent")
+			return nil, errors.Wrap(err, "can't fetch patch intent")
 		}
 		patchDoc = intent.NewPatch()
 		if patchDoc == nil {
-			return errors.New("patch is missing")
+			return nil, errors.New("patch is missing")
 		}
 
 		var projectRef *model.ProjectRef
 		projectRef, err = model.FindOneProjectRefByRepoAndBranchWithPRTesting(patchDoc.GithubPatchData.BaseOwner,
 			patchDoc.GithubPatchData.BaseRepo, patchDoc.GithubPatchData.BaseBranch)
 		if err != nil {
-			return errors.Wrap(err, "can't fetch project ref")
+			return nil, errors.Wrap(err, "can't fetch project ref")
 		}
 		if projectRef == nil {
-			return errors.New("can't find project ref")
+			return nil, errors.New("can't find project ref")
 		}
 
-		status.URLPath = fmt.Sprintf("/waterfall/%s", projectRef.Identifier)
+		status.URL = fmt.Sprintf("%s/waterfall/%s", j.urlBase, projectRef.Identifier)
 		status.Context = "evergreen"
-		status.State = githubStatusFailure
+		status.State = message.GithubStateFailure
 		status.Description = "project config was invalid"
 
-	} else if j.UpdateType == githubUpdateTypeBuild {
-		var b *build.Build
-		b, err = build.FindOne(build.ById(j.FetchID))
-		if err != nil {
-			return err
-		}
-		if b == nil {
-			return errors.New("can't find build")
-		}
-
-		patchVersion = b.Version
-		status.Context = fmt.Sprintf("evergreen/%s", b.BuildVariant)
-		status.Description = trigger.TaskStatusToDesc(b)
-		status.URLPath = fmt.Sprintf("/build/%s", b.Id)
-
-		switch b.Status {
-		case evergreen.BuildSucceeded:
-			status.State = githubStatusSuccess
-
-		case evergreen.BuildFailed:
-			status.State = githubStatusFailure
+	} else if j.UpdateType == githubUpdateTypeNewPatch {
+		status.URL = fmt.Sprintf("%s/version/%s", j.urlBase, j.FetchID)
+		status.Context = "evergreen"
+		status.State = message.GithubStatePending
+		status.Description = "preparing to run tasks"
 
-		default:
-			return errors.New("build status is pending; refusing to update status")
-		}
+	} else if j.UpdateType == githubUpdateTypeRequestAuth {
+		status.URL = fmt.Sprintf("%s/patch/%s", j.urlBase, j.FetchID)
+		status.Context = "evergreen"
+		status.Description = "patch must be manually authorized"
+		status.State = message.GithubStateFailure
 	}
```

Comment:
You removed the default case, so this no longer returns early.

Reply:
As discussed in person, this is fine